utils/http.go

```diff
@@ -163,3 +166,22 @@ func BuildURL(r *rest.Request, template string, params map[string]string) *url.U
 
 	return url
 }
+
+// ApiErr wraps errors returned by our APIs
+type ApiErr struct {
+	Error string `json:"error"`
+}
+
+func ParseApiErr(r *http.Response) (*ApiErr, error) {
```

Comment:
IMO it's better to have `func ParseApiErr(source io.Reader) error`. Then the caller would either check with `IsApiErr()` or convert to `ApiError` if needed.